### ISSUE TITLE
overrides must precede `--projects` and `--targets` in `run-many` and `affected` commands

### ISSUE THREAD

**Issue by Phault** [nrwl/nx#14002]
### Current Behavior

After updating from v14.6.4 to v15.4.0, running `nx affected --target test --ci` now tries to invoke the targets `test` and `--ci`. The same happens with `nx run-many --target test` or `nx run-many --target test --ci`, where the latter will complain that no project called `--ci` exists.

### Expected Behavior

`--ci` should be treated as an override like in v14.6.4 and earlier. Especially since the [CI documentation](https://nx.dev/recipes/ci/monorepo-ci-github-actions) still has overrides last. 

### Steps to Reproduce

1.

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node : 16.16.0
   OS   : darwin arm64
   pnpm : 7.19.0
   
   nx : 15.4.0
   @nrwl/angular : Not Found
   @nrwl/cypress : 15.4.0
   @nrwl/detox : Not Found
   @nrwl/devkit : 15.4.0
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.4.0
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 15.4.0
   @nrwl/js : 15.4.0
   @nrwl/linter : 15.4.0
   @nrwl/nest : Not Found
   @nrwl/next : 15.4.0
   @nrwl/node : Not Found
   @nrwl/nx-cloud : 15.0.2
   @nrwl/nx-plugin : 15.4.0
   @nrwl/react : 15.4.0
   @nrwl/react-native : Not Found
   @nrwl/rollup : 15.4.0
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : 15.4.0
   @nrwl/webpack : 15.4.0
   @nrwl/workspace : 15.4.0
   @nrwl/vite : 15.4.0
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   	 nx-bitbucket-server
   	 nx-github
   	 nx-vercel
   ---------------------------------------
   Community plugins:
   	 @jscutlery/semver: 2.29.3
   	 ngx-deploy-npm: 4.3.10
```

### Additional Information

This has appears to have broken Nx Cloud v15.0.2 agents, since even if you move the overrides first in the main job, the executing agent will spit out the error: `Error: No projects matching: --ci`.

**Comment by sebastiandg7** [comment]
Also happening here. Here you can find an example of a CI run failing due to this: https://github.com/nxkit/nxkit/actions/runs/3778743660/jobs/6425512670#step:16:80

```bash
>  NX   Report complete - copy this into the issue template

   Node : 18.12.1
   OS   : darwin arm64
   pnpm : 7.14.2
   
   nx : 15.4.1
   @nrwl/angular : Not Found
   @nrwl/cypress : Not Found
   @nrwl/detox : Not Found
   @nrwl/devkit : 15.4.1
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.4.1
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 15.4.1
   @nrwl/js : 15.4.1
   @nrwl/linter : 15.4.1
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : 15.0.2
   @nrwl/nx-plugin : 15.4.1
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/rollup : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : Not Found
   @nrwl/webpack : Not Found
   @nrwl/workspace : 15.4.1
   @nrwl/vite : Not Found
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
         @nxkit/style-dictionary
         @nxkit/playwright
   ---------------------------------------
   Community plugins:
         lerna: 6.2.0
```

**Comment by BigGillyStyle** [comment]
I reverted to NX 15.3.3 without any other changes, and this solved my related issue

**Comment by StalkAltan** [comment]
Hi there, thanks for reporting this. We are taking a look. 

As @BigGillyStyle mentioned, the best course of action currently is to stick to 15.3.3 and we will follow up with updates here.

**Comment by vsavkin** [comment]
Hey folks. Sorry about that. This has been fixed in 15.4.2.
